doc/guides/networking/net-stack-architecture.rst

```diff
@@ -0,0 +1,159 @@
+.. _network_stack_architecture:
+
+Network Stack Architecture
+##########################
+
+High level overview of the network stack
+****************************************
+
+.. figure:: zephyr_netstack_overview.svg
+    :align: left
+    :alt: Overview of the network stack architecture
+    :figclass: align-center
+
+    Network stack overview
+
+*The network stack is layered and it consists of the following parts:*
```

Comment:
This should not be italic IMO. To make all of this look better I would have a few paragraphs here with some kind of executive summary, right now we jump into the bulleted details without any introduction and it does not look good.

Reply:
> This should not be italic IMO

Yeah, we can put what ever style here. There are endless possibilities to tweak this document. Lets see David's comments first, and I will send new version.

I try to come up some summary text here for executives, I just expect that no executive will be reading this document.

Reply:
> summary text here for executives

;-) I'm sure @nashif meant a different thing. I would reply that executive summary is in bolded titles of bullet points. But I'm still saddened that my good suggestion to described *each* layer of the stack clearly wasn't taken, so I don't.

Reply:
There is already a overview document that describes in bulleted list things about network stack. I will try to come up some introduction chapter here for architecture doc.

Reply:
> I try to come up some summary text here for executives, I just expect that no executive will be reading this document.

i mean an introduction in general terms about the overall architecture, not for executives and managers. Jumping into bullets directly does not look nice.

Reply:
Yeah, I understood that :)